test/ConfigTest.php

```diff
@@ -12,104 +12,116 @@
 use function uniqid;
 
 /**
- * @coversDefaultClass Laminas\Di\Config
+ * @coversDefaultClass \Laminas\Di\Config
  */
 class ConfigTest extends TestCase
 {
     /** @var array */
-    private $fixture;
+    private $fixture = [];
 
     protected function setUp(): void
     {
         parent::setUp();
         $this->fixture = include __DIR__ . '/_files/sample-config.php';
     }
 
-    public function testGetConfiguredTypeName()
+    public function testGetConfiguredTypeName(): void
     {
         $config = new Config($this->fixture);
-        $this->assertEquals(['Foo', 'Bar'], $config->getConfiguredTypeNames());
+        $this->assertEquals([TestAsset\Config\SomeClass::class, 'SomeAlias'], $config->getConfiguredTypeNames());
```

Comment:
All changes like this were necessary since we do now type with `class-string` explicitly. This was caused especially by the `typeOf` config key which always had to be an existing interface or class name (See [docs](https://docs.laminas.dev/laminas-di/v3/config/)).

It just revealed a logical bug that slipped due to lax typing.